### PR TITLE
cargo fmt

### LINKED ISSUE #19: Enforce rustfmt
The current codebase is almost `cargo fmt` friendly. And diff for enforcing `cargo fmt` is like 15 lines.  I think it provides a cleaner codebase and we have good experience with `rust-miniscript` using `rustfmt`. If there are no NACKs, I can raise a PR to fix the current formatting and to enforce it in the CI.

```
Diff in /home/sanket1729/rust-secp256k1-zkp/src/zkp/pedersen.rs at line 80:
         value: u64,
         generator: Generator,
     ) -> Self {
-        PedersenCommitment::new(secp, value, ZERO_TWEAK ,generator)
+        PedersenCommitment::new(secp, value, ZERO_TWEAK, generator)
     }
 
     pub(crate) fn as_inner(&self) -> &ffi::PedersenCommitment {
Diff in /home/sanket1729/rust-secp256k1-zkp/src/zkp/pedersen.rs at line 376:
         use serde_test::{assert_tokens, Token};
 
         let commitment = PedersenCommitment::from_slice(&[
-            9,7,166,63,171,227,228,157,87,19,233,218,252,171,254,202,
-            228,138,19,124,26,29,131,42,33,212,151,151,89,0,135,201,254,
-        ]).unwrap();
+            9, 7, 166, 63, 171, 227, 228, 157, 87, 19, 233, 218, 252, 171, 254, 202, 228, 138, 19,
+            124, 26, 29, 131, 42, 33, 212, 151, 151, 89, 0, 135, 201, 254,
+        ])
+        .unwrap();
 
         assert_tokens(
             &commitment.readable(),
Diff in /home/sanket1729/rust-secp256k1-zkp/src/zkp/pedersen.rs at line 385:
-            &[Token::Str("0907a63fabe3e49d5713e9dafcabfecae48a137c1a1d832a21d49797590087c9fe")]
+            &[Token::Str(
+                "0907a63fabe3e49d5713e9dafcabfecae48a137c1a1d832a21d49797590087c9fe",
+            )],
         );
 
         assert_tokens(
Diff in /home/sanket1729/rust-secp256k1-zkp/src/zkp/pedersen.rs at line 389:
             &commitment.compact(),
-            &[Token::Bytes(
-                &[
-                    9,7,166,63,171,227,228,157,87,19,233,218,252,171,254,202,
-                    228,138,19,124,26,29,131,42,33,212,151,151,89,0,135,201,254,
-                ]
-            )]
+            &[Token::Bytes(&[
+                9, 7, 166, 63, 171, 227, 228, 157, 87, 19, 233, 218, 252, 171, 254, 202, 228, 138,
+                19, 124, 26, 29, 131, 42, 33, 212, 151, 151, 89, 0, 135, 201, 254,
+            ])],
         );
     }
 
Diff in /home/sanket1729/rust-secp256k1-zkp/src/zkp/rangeproof.rs at line 376:
         );
 
         assert!(opening.message.starts_with(message));
-        assert!(opening.message.ends_with(&vec![0; opening.message.len() - message.len()]));
+        assert!(opening
+            .message
+            .ends_with(&vec![0; opening.message.len() - message.len()]));
     }
 }
 

```

### PR DESCRIPTION
**Pull request by sanket1729**
Fixes #19. As mentioned the diff is minimal and would be really handy in the long run.

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -93,6 +93,11 @@ jobs:
       - name: Pin cc if rust 1.29
         if: matrix.rust == '1.29.0'
         run: cargo generate-lockfile --verbose && cargo update -p cc --precise "1.0.41" --verbose
+      - name: rustfmt on stable
+        if: matrix.rust == 'stable'
+        env:
+          DO_LINT: true
+        run: cargo generate-lockfile --verbose && cargo update -p cc --precise "1.0.41" --verbose
```
Comment (thomaseizinger) [inline .github/workflows/rust.yml:100]: I guess you meant to call `test.sh` here?

**Review by thomaseizinger — CHANGES_REQUESTED** [review]
concept ACK but CI script needs to be fixed

File: .github/workflows/rust.yml (hunk 1 above)
Comment (sanket1729) [reply .github/workflows/rust.yml]: Yes, indeed :facepalm:. Sorry about that. No more 3am coding more me.

**Comment by thomaseizinger** [comment]
Seems like not everything is properly formatted yet?

**Review by thomaseizinger — APPROVED** [review]
ACK d670e5d4553abed05d24808eae44f40a36572fcc

**Review by apoelstra — APPROVED** [review]
utack

**Pull request merged**